### ISSUE TITLE
Disposing item in RemoveItem method

### ISSUE THREAD

**Issue by olliketola** [marklauter/pool#1]
Should this if statement be without "not" condition? Now it states that if  item is not disposable then try to dispose it. Or am I missing something here?
```
    [SuppressMessage("IDisposableAnalyzers.Correctness", "IDISP007:Don't dispose injected", Justification = "it's all private to pool")]
    private void RemoveItem(TPoolItem item)
    {
        lock (this)
        {
            --ItemsAllocated;
        }

        if (!IsPoolItemDisposable)
        {
            (item as IDisposable)?.Dispose();
        }
    }
```
https://github.com/marklauter/pool/blob/1307471c2ecb8470b9fe53a8e5e4c008e9ab8c29/src/Pool/Pool%7BTPoolItem%7D.cs#L270C1-L273C10

**Comment by marklauter** [comment]
Hi, sorry for the late reply. I was out with covid for a few weeks. I'll review your comment this morning.

**Comment by marklauter** [comment]
Thank you for the issue report. It's fixed and will be included in release v4.0.2.
